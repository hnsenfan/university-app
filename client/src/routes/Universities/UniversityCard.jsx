import { useCallback, useState } from 'react'
import PropTypes from 'prop-types'

import { END_POINTS } from '../../api'

import ChevronRight from '../../components/Icons/General/ChevronRight'
import Star from '../../components/Icons/General/Star'

const UniversityCard = ({ uniObject, isFavouriteProps }) => {
  const [isFavourite, setFavourite] = useState(false)

  const handleAddFavourite = useCallback(async() => {
    await END_POINTS.patchFavouriteUniversities(uniObject)
    setFavourite(true)
  })

  return (
    <div className='d-flex flex-column radius-m p-m bg-white mv-l'>
      <div className='d-flex justify-content-between'>
        <label className='text-15 text-medium'>{ uniObject.name }</label>
        <span className='text-medium-grey'>{ uniObject.alpha_two_code }</span>
      </div>
      <div className='d-flex align-items-center justify-content-between mt-s'>
        <label className='text-medium-grey'>{uniObject.country}</label>
        { isFavourite || isFavouriteProps
          ? <Star />
          : <button className='btn-primary fav-button' onClick={handleAddFavourite}>Add as Favourite</button>
        }
      </div>
      <hr className='divider' />
      <div className='d-flex flex-wrap-wrap'>
        { uniObject.web_pages.map((site, index) =>
          <a key={index} className='d-flex align-items-center site-box mt-xs' href={site} target='_blank' rel='noreferrer'>
            <div className='mr-s'>{ site }</div>
            <ChevronRight />
          </a>
        )}
      </div>
    </div>
  )
}

UniversityCard.propTypes = {
  isFavouriteProps: PropTypes.bool,
  uniObject: PropTypes.object,
}

export default UniversityCard