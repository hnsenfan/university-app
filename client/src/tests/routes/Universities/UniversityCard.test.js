import { shallow } from 'enzyme'
import UniversityCard from '../../../routes/Universities/UniversityCard'

test('should test UniversityCard component with default state of empty object', () => {
  const wrapper = shallow(<UniversityCard uniObject={{}} />)
  expect(wrapper).toMatchSnapshot()
 })

 test('should test to render UniversityCard component with sample response', () => {
  const wrapper = shallow(
    <UniversityCard
      uniObject={{
      "domains": ["usmf.md"],
      "web_pages": ["http://www.usmf.md/"],
      "name": "Moldova State University of Medicine and Pharmacy \"N. Testemitsanu\"",
      "alpha_two_code": "MD",
      "state-province": null,
      "country": "Moldova, Republic of"
    }} />
  )
  expect(wrapper).toMatchSnapshot()
 })