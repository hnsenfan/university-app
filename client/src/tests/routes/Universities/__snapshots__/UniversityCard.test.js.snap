// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test UniversityCard component with default state of empty object 1`] = `
<div
  className="d-flex flex-column radius-m p-m bg-white mv-l"
>
  <div
    className="d-flex justify-content-between"
  >
    <label
      className="text-15 text-medium"
    />
    <span
      className="text-medium-grey"
    />
  </div>
  <div
    className="d-flex align-items-center justify-content-between mt-s"
  >
    <label
      className="text-medium-grey"
    />
    <button
      className="btn-primary fav-button"
      onClick={[Function]}
    >
      Add as Favourite
    </button>
  </div>
  <hr
    className="divider"
  />
  <div
    className="d-flex flex-wrap-wrap"
  />
</div>
`;

exports[`should test to render UniversityCard component with sample response 1`] = `
<div
  className="d-flex flex-column radius-m p-m bg-white mv-l"
>
  <div
    className="d-flex justify-content-between"
  >
    <label
      className="text-15 text-medium"
    >
      Moldova State University of Medicine and Pharmacy "N. Testemitsanu"
    </label>
    <span
      className="text-medium-grey"
    >
      MD
    </span>
  </div>
  <div
    className="d-flex align-items-center justify-content-between mt-s"
  >
    <label
      className="text-medium-grey"
    >
      Moldova, Republic of
    </label>
    <button
      className="btn-primary fav-button"
      onClick={[Function]}
    >
      Add as Favourite
    </button>
  </div>
  <hr
    className="divider"
  />
  <div
    className="d-flex flex-wrap-wrap"
  >
    <a
      className="d-flex align-items-center site-box mt-xs"
      href="http://www.usmf.md/"
      key="0"
      rel="noreferrer"
      target="_blank"
    >
      <div
        className="mr-s"
      >
        http://www.usmf.md/
      </div>
      <ChevronRight />
    </a>
  </div>
</div>
`;
